apps/data-pipeline/sample-program-scraper/src/index.ts

```diff
@@ -199,73 +229,42 @@ async function lookupCourseId(
   }
 }
 
+/*
+  Validates a single course code and returns a discriminated union.
+  Returns courseId type if found in database, unknown type otherwise.
+*/
+async function validateCourseCode(
+  db: ReturnType<typeof database>,
+  code: string,
+): Promise<{ type: "courseId"; value: string } | { type: "unknown"; value: string }> {
```

Comment:
I think this should be outlined too. It's one of the most likely points to be changed in the future.